containers/hawkular-htpasswd/src/main/java/org/hawkular/openshift/htpasswd/CreatePassword.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2014-2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.openshift.htpasswd;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.hawkular.openshift.auth.PasswordManager;
+
+public class CreatePassword {
+
+    private static final String SHA256_ALGORITHM = "pbkdf2-sha256";
+    private static final String SHA512_ALGORITHM = "pbkdf2-sha512";
+
+    private String algorithm;
+    private String fileName;
+    private String password;
+    private String username;
+    private boolean created;
+
+    private PasswordManager passwordManager = new PasswordManager();
+
+    public CreatePassword() {
+        password = System.getProperty("htpasswd.password", "");
+        username = System.getProperty("htpasswd.username", "");
+        algorithm = System.getProperty("htpasswd.algorithm", SHA256_ALGORITHM);
+        fileName = System.getProperty("htpasswd.file", "");
+        created = Boolean.getBoolean("htpasswd.created");
```

Comment:
Wouldn't `htpasswd.overwrite` be a more appropriate name? IIUC if this is `true` then an existing file should be overwritten, if `false` then a line should be appended, right?

Reply:
Is right, I took the name from htpasswd utility, but this is a better name.